ci/lava/conftest.py

```diff
@@ -7,85 +7,108 @@
-                report.nodeid.replace(" ", "_")
-            )
-            return report.outcome, "*", lava
-
-    if report.failed:
-        lava = "<LAVA_SIGNAL_TESTCASE TEST_CASE_ID={} RESULT=fail>".format(
-            report.nodeid.replace(" ", "_")
-        )
-        return report.outcome, "*", lava
+from helpers import download_from_url, get_dut_address, ExecuteHelper
 
 
 def pytest_addoption(parser):
     """Add option parsing to pytest."""
+    # Common parameters
+    parser.addoption("--venv", action="store", default="/tmp/venv")
     parser.addoption("--debug-output", action="store_true")
     parser.addoption("--device", action="store", default="none")
     parser.addoption("--dut", action="store", default="auto")
     parser.addoption(
         "--dut-download-dir", action="store", default="/tmp/pytest"
     )
-    parser.addoption("--dut-tutorials-dir", action="store", default="/scratch")
     parser.addoption(
         "--host-download-dir", action="store", default="/tmp/pytest"
     )
+
+    # Tutorial tests parameters
+    parser.addoption("--dut-tutorials-dir", action="store", default="/scratch")
     parser.addoption(
         "--host-tutorials-dir", action="store", default="/tmp/tutorials"
     )
-    parser.addoption("--local-conf-file", action="store", default="")
     parser.addoption("--payload-version", action="store", default="1")
-    parser.addoption("--venv", action="store", default="/tmp/venv")
+
+    # local.conf tests parameters
+    parser.addoption("--local-conf-file", action="store", default="")
+
+    # Boot component update test parameters
+    parser.addoption(
+        "--update-component-name",
+        action="store",
+        choices=["wks_bootloader1", "wks_bootloader2", "kernel"],
+    )
+    parser.addoption("--update-payload-url", action="store")
+    parser.addoption(
+        "--update-method",
+        action="store",
+        default="mbl-cli",
+        choices=["pelion", "mbl-cli"],
+    )
 
 
-def _download_from_url(url):
+def pytest_configure(config):
+    """Registering custom markers"""
+    config.addinivalue_line("markers", "pelion: mark a test to be run when "
+        "the update method is pelion"
+    )
+    config.addinivalue_line("markers", "mbl_cli: mark a test to be run when "
+        "the update method is mbl-cli"
+    )
 
-    filename = None
 
-    if url:
+def pytest_collection_modifyitems(config, items):
+    """Select tests depending the update-method passed as argument."""
+    update_method = config.getoption("--update-method")
+    skip = pytest.mark.skip(
+        reason="update-method {} selected".format(update_method)
+    )
+    for item in items:
+        markers = set([m.name for m in item.iter_markers()])
+        # We skip tests which don't match the update_method
+        # mbl-cli is replaced with mbl_cli
+        if (
+            update_method.replace("-", "_") not in markers
+            and not len(markers) == 0
```

Comment:
If you're checking the markers set isn't empty you should just be able to do `and markers` rather than `not len(markers) == 0` which is harder to read